### PR TITLE
Remove the usage of CAutoLock

### PR DESCRIPTION
**Pull request by miguelgaio**
CAutoLock hide a basic std::lock_guard usage.

Replace all occurences of CAutoLock by its standard definition
std::lock_guard<std::mutex>

Signed-off-by: Miguel Gaio <miguel.gaio@intel.com>
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/228%23issuecomment-140450865%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/228%23issuecomment-140729411%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/228%23issuecomment-140730583%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/228%23issuecomment-140730881%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/228%23discussion_r39484800%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/228%23discussion_r39484938%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/228%23discussion_r39524483%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/228%23discussion_r39524484%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/228%23discussion_r39524485%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/228%23issuecomment-140450865%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3A%2B1%3A%20%20%40dawagner%20%40OznOg%20please%20review%22%2C%20%22created_at%22%3A%20%222015-09-15T16%3A23%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40cc6565%20%40louizo%20Please%20review.%20I%20will%20not%20merge%20patches%20without%20at%20least%202%20reviews%20%28excluding%20author%29.%22%2C%20%22created_at%22%3A%20%222015-09-16T12%3A38%3A06Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-09-16T12%3A44%3A06Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9692938%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cc6565%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%20%22%2C%20%22created_at%22%3A%20%222015-09-16T12%3A45%3A42Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%2022793fc77b1eba2b2be1fa0905164fe6e38a8a0a%20parameter/ParameterHandle.cpp%2015%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/228%23discussion_r39524483%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-09-15T15%3A27%3A46Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/ParameterHandle.cpp%3AL81-88%22%7D%2C%20%22Pull%2022793fc77b1eba2b2be1fa0905164fe6e38a8a0a%20parameter/ParameterHandle.cpp%208%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/228%23discussion_r39484800%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%20would%20add%20a%5Cr%5Cn%60%60%60%20cpp%5Cr%5Cnusing%20std%3A%3Amutex%3B%5Cr%5Cnusing%20std%3A%3Alock_guard%3B%5Cr%5Cn%60%60%60%5Cr%5CnThat%20would%20avoid%2040%20%60std%3A%3A%60%20repetition%20%3A%29%22%2C%20%22created_at%22%3A%20%222015-09-15T08%3A17%3A51Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-09-15T15%3A27%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/ParameterHandle.cpp%3AL33-41%22%7D%2C%20%22Pull%2022793fc77b1eba2b2be1fa0905164fe6e38a8a0a%20parameter/ParameterMgr.cpp%2021%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/228%23discussion_r39484938%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22There%20is%20not%20much%20repetition%20here%2C%20the%20%60using%60%20trick%20would%20not%20make%20us%20gain%20any%20line.%22%2C%20%22created_at%22%3A%20%222015-09-15T08%3A19%3A27Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-09-15T15%3A27%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/ParameterMgr.cpp%3AL739-746%22%7D%7D%2C%20%22approved%22%3A%20%7B%22https%3A//github.com/krocard%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/krocard'></a>

- [x] <a href='#crh-comment-Pull 22793fc77b1eba2b2be1fa0905164fe6e38a8a0a parameter/ParameterHandle.cpp 8'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/228#discussion_r39484800'>File: parameter/ParameterHandle.cpp:L33-41</a></b>
- <a href='https://github.com/krocard'></a> I would add a
``` cpp
using std::mutex;
using std::lock_guard;
```
That would avoid 40 `std::` repetition :)
- [x] <a href='#crh-comment-Pull 22793fc77b1eba2b2be1fa0905164fe6e38a8a0a parameter/ParameterMgr.cpp 21'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/228#discussion_r39484938'>File: parameter/ParameterMgr.cpp:L739-746</a></b>
- <a href='https://github.com/krocard'></a> There is not much repetition here, the `using` trick would not make us gain any line.
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/228#issuecomment-140450865'>General Comment</a></b>
- <a href='https://github.com/krocard'></a> :+1:  @dawagner @OznOg please review
- <a href='https://github.com/krocard'></a> @cc6565 @louizo Please review. I will not merge patches without at least 2 reviews (excluding author).

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/228?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/228?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/228?approve=0'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/228'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dawagner** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/01org/parameter-framework/pull/228'>Refresh</a>.<a href='#crh-update-none'></a>

File: parameter/ParameterHandle.cpp [hunk 1]
```diff
@@ -80,7 +81,7 @@ bool CParameterHandle::setAsBoolean(bool bValue, string& strError)
         return false;
     }
     // Ensure we're safe against blackboard foreign access
-    CAutoLock autoLock(_pParameterMgr->getBlackboardMutex());
+    std::lock_guard<std::mutex> autoLock(_pParameterMgr->getBlackboardMutex());
```
Comment (krocard) [inline parameter/ParameterHandle.cpp]: I would add a
```
using std::lock;
using std::lock_guard;
```
That would avoid 40 `std::` repetition :)

File: parameter/ParameterHandle.cpp [hunk 2]
```diff
@@ -33,7 +33,8 @@
 #include "Subsystem.h"
 #include <assert.h>
 #include "ParameterMgr.h"
-#include "AutoLock.h"
+
+#include <mutex>
 
 using std::string;
```
Comment (krocard) [inline parameter/ParameterHandle.cpp]: I would add a
``` cpp
using std::lock;
using std::lock_guard;
```
That would avoid 40 `std::` repetition :)

File: parameter/ParameterMgr.cpp [hunk 3]
```diff
@@ -739,7 +739,7 @@ void CParameterMgr::applyConfigurations()
     LOG_CONTEXT("Configuration application request");
 
     // Lock state
-    CAutoLock autoLock(getBlackboardMutex());
+    std::lock_guard<std::mutex> autoLock(getBlackboardMutex());
```
Comment (krocard) [inline parameter/ParameterMgr.cpp]: There is not much repetition here, the `using` trick would not make us gain any line.

File: parameter/ParameterHandle.cpp (hunk 1 above)
Comment (miguelgaio) [inline parameter/ParameterHandle.cpp]: :+1:<a href='#crh-update-none'></a>

File: parameter/ParameterHandle.cpp (hunk 2 above)
Comment (miguelgaio) [inline parameter/ParameterHandle.cpp]: :+1:<a href='#crh-update-none'></a>

File: parameter/ParameterMgr.cpp (hunk 3 above)
Comment (miguelgaio) [inline parameter/ParameterMgr.cpp]: :+1:<a href='#crh-update-complete'></a>

**Comment by krocard** [comment]
:+1:

**Comment by krocard** [comment]
@cc6565 @louizo Please review. I will not merge patches without at least 2 review (excluding author).

**Comment by cc6565** [comment]
:+1:

**Comment by krocard** [comment]
:shipit:

**Pull request merged**